### DISCUSSION TITLE
Can Isar replace json_serializable?

### DISCUSSION THREAD

**Discussion by SuricateTrading** [isar/isar#1167] [Help]
I use the package json_serializable to convert json Maps to objects,
Is Isar able to do this too?
I looked into the generated file, but the "FooDeserialize" method requires an IsarReader.
It would be great if Isar could deserialize without the need of anything else but a json map, similar to the json_serializable package.

If not, I would need to annotate every field twice, both for Isar and json_serializable. This would make all data classes extremely ugly.

**Comment by frank06** [comment]
An alternative would be loading JSON into Isar and then querying.

```dart
collection.isar.writeTxn(() => collection.importJson([data]));
```

**Reply by frank06** [reply]
Or use https://github.com/simc/crimson/ , although with the author you never know how long is going to be maintained.
